cypress/e2e/gui/customers.cy.js

```diff
@@ -0,0 +1,187 @@
+      cy.contains('button', 'Back').click();
+
+      cy.contains('p', 'Below is our customer list.').should('be.visible');
+    });
+  });
+
+  describe('Empty State', () => {
+    beforeEach(() => {
+      cy.setCookie('cookieConsent', 'accepted');     
+      cy.visit('/');
+      cy.intercept('GET', `${CUSTOMERS_URL}?page=1&limit=10&size=All&industry=All`, { fixture: 'customers/emptyCustomers' }).as('getEmptyCustomers');
+      cy.get('[data-testid="size-filter"]').select('All');
+      cy.wait('@getEmptyCustomers');
+    });
+
+    it('It shows the image of an empty box and the text "No customers available." when there are no customers in the database', () => {
+      cy.get('svg[title="image of an empty box"]').should('be.visible');
+      cy.contains('span', 'No customers available.').should('be.visible');
+    });
+
+    it('It disables the name text input field when there are no customers in the database', () => {
+      cy.get('#name').should('be.disabled');
+    })
+  });
+
+  describe('Messenger Form', () => {
+    beforeEach(() => {
+      cy.setCookie('cookieConsent', 'accepted');     
+      cy.visit('/');
+    });
+    
+    it('It shows and hides a success message when successfully submitting the messenger form', () => {
+      const messengerData = {
+        name: 'Matheus',
+        email: 'matheus@gmail.com',
+        message: 'Hello, I need help with something.',
+      };
+      cy.gui_submitMessenger(messengerData);
+
+      cy.get('[class^="Messenger_success"]').should('be.visible').and('have.text', 'Your message has been sent.');
+      cy.get('[class^="Messenger_success"]').should('not.exist');
+    });
+  });
+
+  describe('Filters', () => {
+    beforeEach(() => {
+      cy.setCookie('cookieConsent', 'accepted');     
+      cy.visit('/');
+    });
+  
+    it('It keeps the filters when coming back from the customer details view', () => {
+      const clientFilterData = {
+        size: 'Small',
+        industry: 'Technology',
+      };
+      cy.gui_filterClients(clientFilterData);
+      cy.get('button[aria-label^="View company:"]').first().click();
+      cy.contains('button', 'Back').click();
+
+      cy.findByTestId('size-filter').should('have.value', clientFilterData.size);
+      cy.findByTestId('industry-filter').should('have.value', clientFilterData.industry);
+    });
+
+    it('It persists the limit of items per page in the local storage when changing the limit', () => {
+      const paginationLimit = '20';
+      cy.get('[name="pagination-limit"]').select(paginationLimit);
+
+      cy.window().then((win) => {
+        expect(win.localStorage.getItem('paginationLimit')).to.eq(paginationLimit);
+      });
+    });
+
+    describe('Filter by size', () => {
+      it('It filters the customers by small size', () => {
+        cy.intercept('GET', `${CUSTOMERS_URL}?page=1&limit=10&size=Small&industry=All`, { fixture: 'customers/smallCustomers' }).as('getSmallCustomers');
```

Comment:
Recomendo quebrar linha aqui para ajudar na legibilidade.

```suggestion
        cy.intercept(
          'GET', 
          ${CUSTOMERS_URL}?page=1&limit=10&size=Small&industry=All`,
          { fixture: 'customers/smallCustomers' }
        ).as('getSmallCustomers');
```

Que tal?

Reply:
O mesmo vale para os próximos testes.